src/index.ts

```diff
@@ -1,10 +1,13 @@
-import { Users } from './services/users';
-import { Logger } from './services/logger';
+import type { IoCResources, User } from './types';
+import { createIoCContainer } from './ioc';
+import IoCContainer from 'ioc-lite';
 
-import type { User, ApiConfig } from './types';
+let ioc: IoCContainer<IoCResources>;
```

Comment:
Is there a reason not to create an ioc container right here?

`const ioc = createIoCContainer();`

Reply:
Did like so first time and I was getting error something like "config was not found", tried once again and it is working, weird.